compiler/rustc_hir_analysis/src/collect.rs

```diff
@@ -1025,7 +1025,13 @@ fn adt_def(tcx: TyCtxt<'_>, def_id: LocalDefId) -> ty::AdtDef<'_> {
 
     let is_anonymous = item.ident.name == kw::Empty;
     let repr = if is_anonymous {
-        tcx.adt_def(tcx.local_parent(def_id)).repr()
+        let parent = tcx.local_parent(def_id);
+        if matches!(tcx.hir_node_by_def_id(parent), Node::Item(_)) {
+            tcx.adt_def(tcx.local_parent(def_id)).repr()
+        } else {
+            tcx.dcx().span_delayed_bug(item.span, "anonymous field inside non struct/union");
+            tcx.repr_options_of_def(def_id.to_def_id())
```

Comment:
Since this is always empty since anon adts can't have their own `repr` afaik, this feels a bit misleading / confusing. I would just go for:

```suggestion
            ty::ReprOptions::default()
```